docs/developer/using_devel_codes.md

```diff
@@ -0,0 +1,136 @@
+# Using Devel Codes
+
+This document is intended to describe the use of devel codes for
+developers to speed up implementation of experimental parameters.
+
+## What are devel codes
+
+Devel codes are handy tools for developers to control features while
+they are not ready for public release. Devel codes are read from the
+`devel_code` block of the `.param` file.
+
+Devel codes are much simpler to use than keywords, however they do not
+have the rigorous type/value checking or unit conversion that keywords
+do. Nor are they searchable and do not provide any help. In order to
+use devel codes you have to explicitly know how to call them, the
+values they should take and what they do.
+
+The devel code block is an arbitrary string which can be used to
+provide parameters to a function, however, the devel code block also
+has several useful standard structural elements to make it more useful
+than just a string one has to parse themselves.
+
+## Using devel codes
+
+### Devel code sub-blocks
+
+Devel codes can be more explicitly subdivided into sub-blocks which
+are usually named related to the module or functions they support.
+
+Examples include:
+
+- `MD` - Relating to molecular dynamics options
+- `FIRE` - Relating to the FIRE geometry optimisation method
+- `ELECTRONIC` - Relating to electronic minimisation
+
+However they can be anything the developer wants. Keeping it related
+to the module where they are used is useful for remembering how to
+call them, but is not necessary. Indeed, it may be desirable to have a
```

Comment:
I think at this stage in CASTEP evolution we should make the use of the block compulsory, not optional, to prevent namespace clashes and to help readability. Now that we have the io routines for parsing, it is very simple to do so. No real reason NOT to do so.